code/game/objects/structures/viewscreen.dm

```diff
@@ -124,25 +126,28 @@
 
 /obj/structure/viewscreen_controller
 	name = "viewscreen control panel"
+	desc = "This is a control panel for the viewscreens"
 	icon = 'icons/obj/airlock_machines.dmi'
 	icon_state = "airlock_control_standby"
 
 	density = 0
 	anchored = 1
 
-	var/obj/structure/viewscreen/linked = null
-
-
-/obj/structure/viewscreen_controller/Initialize()
-	for(var/obj/structure/viewscreen/V in world)
-		linked = V
-
-	..()
+/obj/structure/viewscreen_controller/proc/get_area_viewscreens()
+	var/list/screens = list()
+	var/turf/T = get_turf(src)
+	var/area/A = get_area(T)
+	for(var/obj/structure/viewscreen/V in area_contents(A))
+		screens += V
+	return screens
 
 /obj/structure/viewscreen_controller/attack_hand(mob/user)
 	var/dat = "<B>Viewscreen Control Panel</B><HR>"
-	dat += "Viewing Mode: [linked.view_mode ? "Navigation" : "Tactical"]"
-	dat += "<BR><BR><A href=?src=\ref[src];switch=1>Switch View Modes</A>"
+
+	for(var/obj/structure/viewscreen/V in get_area_viewscreens())
```

Comment:
You should not be re checking the area for viewscreen every time someone clicks the controller. Make a list on the controller holding all known screens instead and put a refresh button in the ui.